### PR TITLE
Use a build matrix to check queries

### PR DESCRIPTION
**Pull request by vigoux**
This will allow use to produce parsers artifacts for macos and linux at least for now.

### CODE REVIEW

File: .github/workflows/check_query_files.yml [hunk 1]
```diff
@@ -5,7 +5,10 @@ on: [push, pull_request]
 jobs:
   luacheck:
     name: Check Query Files
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [unbuntu-latest, macos-latest]
```
Comment (vigoux) [inline .github/workflows/check_query_files.yml:11]:
```suggestion
        os: [ubuntu-latest, macos-latest]
```

**Comment by theHamsta** [comment]
Shouldn't this be a separate testing workflow?

I tried to separate Lua style checking from the syntax of the query files. The query files (typo in queries) should be unrelated to OS but further testing should be done for multiple OSes.

**Comment by vigoux** [comment]
Then I think that we need a workflow to just compile the parsers on every platform maybe every day or so.

**Comment by theHamsta** [comment]
I mean having a workflow for tests on every platform (automatically compiles the parsers) and one workflow for the queries. But one big workflow that tests everything would also be ok.

**Comment by vigoux** [comment]
Yeah I see what you mean, but I will to do that in another PR, this one is a mess.

**Pull request closed without merging**